include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -730,25 +730,29 @@ ::reinit_impl (MatrixArray &&mass_matrices_,
   this->derivative_matrix = derivative_matrix;
 
   constexpr unsigned int macro_size = VectorizedArray<Number>::n_array_elements;
-  const unsigned int nm_flat_size
-    = (size > 0)
-      ? (Utilities::fixed_int_power<size,dim>::value
-         * Utilities::fixed_int_power<size,dim>::value * macro_size)
-      : (Utilities::fixed_power<dim>(mass_matrix[0].n_rows())
-         * Utilities::fixed_power<dim>(mass_matrix[0].n_rows()) * macro_size);
-  const unsigned int n_flat_size
-    = (size > 0)
-      ? Utilities::fixed_int_power<size,dim>::value * macro_size
-      : Utilities::fixed_power<dim>(mass_matrix[0].n_rows()) * macro_size;
+  const auto n_rows_max = [&mass_matrix]()
+  {
+    std::size_t n_rows_max {size};
+    if (size == -1)
+      {
+        std::array<std::size_t,dim> n_rows;
+        for (unsigned int d = 0; d < dim; ++d)
+          n_rows[d] = mass_matrix[d].n_rows();
+        n_rows_max = *(std::max_element (n_rows.cbegin(),n_rows.cend()));
```

Comment:
This looks too complicated. Why not simply do the computation as
```
std::size_t n_rows_max = 0;
for (unsigned int d=0; d<dim; ++d)
  n_rows_max = std::max(n_rows_max, mass_matrix[d].n_rows();
```

Reply:
Yes, thanks! In addition, I replaced the `reserve` with `resize` and deleted redundant `resize`s inside the loop, since `vectorized_transpose_and_store` has the control over the actual data size.